lib/mixpanel-node.js

```diff
@@ -148,16 +148,24 @@ var create_client = function(token, config) {
             }
             try {
                 body = JSON.parse(body);
-                if (body.status === 0) {
-                    error = new Error('Mixpanel REST API reported an error: ' + body.error);
-                    console.error(error);
-                    return callback(error);
-                }
             } catch (e) {
-                error = new Error('Mixpanel REST API returned invalid JSON response');
+                error = new Error('Mixpanel REST API returned an invalid JSON response');
+                console.error(error);
+                return callback(error);
+            }
+
+            if (!body || typeof body.status !== 'number') {
```

Comment:
Well, the first if `(!body || typeof body.status !== 'number')` checks that the response is valid, while the second tests if there is an error. If instead I did something like `if (!body || body.status == 1)`, then I would have to make another if inside, to distinguish between an invalid response, and an error response. Am I misunderstanding your question?